frontend/src/app/services/source/source.service.ts

```diff
@@ -40,21 +39,23 @@ export class SourceService {
     source: Source
   ): Observable<Source> {
     return new Observable<Source>((subscriber) => {
-      this.projectService.init(project_slug).subscribe((project) => {
-        this.http
-          .post<Source>(
-            environment.backend_url +
-              '/projects/' +
-              project.name +
-              '/extensions/modelsources/git/create/' +
-              model_slug,
-            source
-          )
-          .subscribe((new_source) => {
-            subscriber.next(new_source);
-            subscriber.complete();
-          });
-      });
+      this.projectService
+        .getProjectBySlug(project_slug)
+        .subscribe((project) => {
+          this.http
+            .post<Source>(
+              environment.backend_url +
+                '/projects/' +
+                project.name +
+                '/extensions/modelsources/git/create/' +
+                model_slug,
+              source
+            )
+            .subscribe((new_source) => {
+              subscriber.next(new_source);
+              subscriber.complete();
+            });
+        });
```

Comment:
Is there a need to fetch the model again (instead of using `this.projectService.project`)?

Reply:
It depends on where the method is used. It’s better to decorrelate, so I change to passing directly the name of the project to the function instead of getting it from the slug. (The function has no idea of its context of use, so it makes no sense that it uses `this.projectService._project`, which is binded to the context.)